src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c

```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
+                        this->actor.textId = 0x1B;
+                        func_8010B720(globalCtx, this->actor.textId);
+                        this->dialogState = 5;
+                        func_800800F8(globalCtx, 0x1F4A, -0x63, NULL, 0);
+                        func_8002DF54(globalCtx, NULL, 8);
+                        this->actionFunc = EnBomBowMan_SetupChooseShowPrize;
+                    }
+                } else {
+                    this->playingAgain = false;
+                    this->actor.textId = 0x85;
+                    func_8010B720(globalCtx, this->actor.textId);
+                    this->dialogState = 5;
+                    this->actionFunc = func_809C41FC;
+                }
+                break;
+            case 1: // No
+                this->playingAgain = false;
+                this->actor.textId = 0x2D;
+                func_8010B720(globalCtx, this->actor.textId);
+                this->dialogState = 5;
+                this->actionFunc = func_809C41FC;
+        }
+    }
+}
+
+void func_809C41FC(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+        if ((this->actor.textId == 0x2D) || (this->actor.textId == 0x85)) {
+            if (Flags_GetSwitch(globalCtx, 0x38)) {
+                Flags_UnsetSwitch(globalCtx, 0x38);
+            }
+        }
+        if (this->minigamePlayStatus == 1) {
+            this->actor.textId = 0x1B;
+            func_8010B720(globalCtx, this->actor.textId);
+            this->dialogState = 5;
+            func_800800F8(globalCtx, 8010, -99, NULL, 0);
+            func_8002DF54(globalCtx, NULL, 8);
+            this->actionFunc = EnBomBowMan_SetupChooseShowPrize;
+        } else {
+            if (this->gameResult == 2) {
+                func_8002DF54(globalCtx, NULL, 7);
+            }
+            this->actionFunc = EnBomBowMan_SetupRunGame;
+        }
+    }
+}
+
+void EnBomBowMan_SetupChooseShowPrize(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    Vec3f accel = { 0.0f, 0.1f, 0.0f };
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f pos;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        pos.x = 148.0f;
+        pos.y = 40.0f;
+        pos.z = 300.0f;
+        EffectSsBomb2_SpawnLayered(globalCtx, &pos, &velocity, &accel, 50, 15);
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_GOODS_APPEAR);
+        this->prizeRevealTimer = 10;
+        this->actionFunc = EnBomBowMan_ChooseShowPrize;
+    }
+}
+
+static Vec3f sPrizePosOffset[] = {
+    { 0.0f, 22.0f, 0.0f }, { 0.0f, 22.0f, 0.0f }, { 0.0f, 8.0f, 0.0f }, { 0.0f, 9.0f, 0.0f }, { 0.0f, -2.0f, 0.0f },
+};
+
+static s16 sPrizeRot[] = { 0x4268, 0x4268, 0xFC18, 0x0000, 0x4268, 0x0000 };
```

Comment:
```suggestion
static s16 sPrizeRot[] = { 0x4268, 0x4268, -0x03E8, 0x0000, 0x4268, 0x0000 };
```
Note: All these values make a lot more sense as decimal, like it's often the case, but the whole hex vs dec debate still needs to be resolved separately